cmd2/cmd2.py

```diff
@@ -1998,18 +1998,29 @@ def onecmd(self, statement: Union[Statement, str]) -> bool:
         if statement.command in self.macros:
             stop = self._run_macro(statement)
         else:
-            func = self.cmd_func(statement.command)
+            command = statement.command
+            func = self.cmd_func(command)
+            func_arg = statement.args
+
+            if not func and self.default_to_shell:
```

Comment:
I very much like changing the logic of how ``default_to_shell`` behavior works so that it directly uses the ``do_shell`` method.  However, I'm not sure I like moving that logic out of the ``do_default`` method.  

The benefit of having it within the ``do_default`` method is that it allows end users to override ``do_default`` however they choose and get reasonable expected behavior.